### ISSUE TITLE
Broken APREC Support

### ISSUE THREAD

**Issue by HaoZeke** [shogun-toolbox/shogun#4420]
# Bug Report
- [x] No similar issues exist
- [x] Replicated on similar systems

## Description
The `make` step fails if ARPREC is installed.

## Configuration

``` bash
 cmake .. -DBUILD_META_EXAMPLES=OFF                     
-- dir='/home/haozeke/Git/Github/machineLearning/shogun/src'
-- dir='/home/haozeke/Git/Github/machineLearning/shogun/build/src'
-- dir='/home/haozeke/Git/Github/machineLearning/shogun/src/gpl'
-- Using system's malloc
-- Could NOT find ViennaCL (missing: VIENNACL_INCLUDE_DIR VIENNACL_ENCODED_VERSION) (Required is at least version "1.5.0")
-- Could NOT find TFLogger (missing: TFLogger_DIR)
-- Could NOT find CPLEX (missing: CPLEX_LIBRARY CPLEX_INCLUDE_DIR) 
-- Found Arpack : /usr/lib/libarpack.so
-- Could NOT find MOSEK (missing: MOSEK_DIR MOSEK_INCLUDE_DIR MOSEK_LIBRARY MOSEK_LIBRARIES) 
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Lzo Library /usr/lib/liblzo2.so
-- Lzo Include Found in /usr/include
-- Could NOT find LPSOLVE (missing: LPSOLVE_LIBRARIES LPSOLVE_INCLUDE_DIR) 
-- Could NOT find ColPack (missing: COLPACK_LIBRARIES COLPACK_LIBRARY_DIR COLPACK_INCLUDE_DIR) 
-- Linker: GNU gold
-- ===================================================================================================================
-- Summary of Configuration Variables
-- The following OPTIONAL packages have been found:

 * CCache
 * GDB
 * Doxygen (required version >= 1.8.6)
 * OpenMP
 * CxaDemangle
 * rxcpp
 * BLAS
 * GLPK
 * ARPACK
 * Threads
 * Protobuf
 * JSON (required version >= 0.11)
 * LibXml2
 * HDF5
 * CURL
 * ZLIB
 * BZip2
 * LibLZMA
 * SNAPPY
 * LZO
 * NLopt
 * ARPREC
 * Sphinx
 * Pandoc

-- The following REQUIRED packages have been found:

 * PythonInterp

-- The following OPTIONAL packages have not been found:

 * ViennaCL (required version >= 1.5.0)
 * TFLogger (required version >= 0.1.0)
 * CPLEX
 * Mosek
 * LpSolve
 * ColPack

-- ===================================================================================================================
-- Integrations
--   OpenCV Integration is OFF 	 enable with -DOpenCV=ON
-- ===================================================================================================================
-- Interfaces
--   Python is OFF 		 enable with -DINTERFACE_PYTHON=ON
--   Octave is OFF 		 enable with -DINTERFACE_OCTAVE=ON
--   Java is OFF 		 enable with -DINTERFACE_JAVA=ON
--   Perl is OFF 		 enable with -DINTERFACE_PERL=ON
--   Ruby is OFF 		 enable with -DINTERFACE_RUBY=ON
--   C# is OFF 			 enable with -DINTERFACE_CSHARP=ON
--   R is OFF 			 enable with -DINTERFACE_R=ON
--   Lua is OFF 		 enable with -DINTERFACE_LUA=ON
--   Scala is OFF 		 enable with -DINTERFACE_SCALA=ON
--   CoreML is OFF 		 enable with -DINTERFACE_COREML=ON
-- ===================================================================================================================
-- To compile shogun type
--   make
-- 
-- To install shogun to /usr/local type
--   make install
-- 
-- or to install to a custom directory
--   make install DESTDIR=/my/special/path
--   (or rerun cmake with -DCMAKE_INSTALL_PREFIX=/my/special/path) to just change the prefix
-- ===================================================================================================================
-- Configuring done
-- Generating done
-- Build files have been written to: /home/haozeke/Git/Github/machineLearning/shogun/build
```

Where examples are turned off due to https://github.com/shogun-toolbox/shogun/issues/4419.

## Error
```bash
Scanning dependencies of target libshogun
[ 74%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/class_list.cpp.o
[ 74%] Built target libshogun
[ 74%] Linking CXX shared library libshogun.so
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(add.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(complex.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(cpr.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(cssx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(div.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(divd.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(divx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(dmc.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(eq.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(expx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(init.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp6old_cwE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(init.o): requires dynamic R_X86_64_PC32 reloc against '_ZTTSt14basic_ifstreamIcSt11char_traitsIcEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(init.o): requires dynamic R_X86_64_PC32 reloc against '_ZN7mp_real3_piE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(inix.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp5mpuu1E' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(log.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(log.o): requires dynamic R_X86_64_PC32 reloc against '_ZNKSt5ctypeIcE8do_widenEc' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(logx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(mdc.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(mul.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(muld.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(mulx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(nint.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp10prec_wordsE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(norm.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(npwx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(pix.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(roun.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(sqrt.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(sqrtx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(sqx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(sub.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(binary_io.o): requires dynamic R_X86_64_PC32 reloc against '_ZSt4cerr' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTVSt9basic_iosIcSt11char_traitsIcEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTVSt15basic_streambufIcSt11char_traitsIcEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTVSt15basic_streambufIcSt11char_traitsIcEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTV17fixed_col_ostreamIcSt11char_traitsIcEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTV17fixed_col_ostreamIcSt11char_traitsIcEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTVNSt7__cxx1115basic_stringbufIcSt11char_traitsIcESaIcEEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTTNSt7__cxx1119basic_ostringstreamIcSt11char_traitsIcESaIcEEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(write.o): requires dynamic R_X86_64_PC32 reloc against '_ZTVNSt7__cxx1115basic_stringbufIcSt11char_traitsIcESaIcEEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(mp_real.o): requires dynamic R_X86_64_PC32 reloc against '_ZSt4cerr' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(debug.o): requires dynamic R_X86_64_PC32 reloc against '_ZSt4cerr' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(dble.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp5mprdxE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(agmx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(angx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(cshx.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(cssn.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(exp.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(infr.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(lconv.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp6mpmcrxE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(npwr.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp10prec_wordsE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(pi.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(rand.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(to_digits.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp5mpbdxE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(read.o): requires dynamic R_X86_64_PC32 reloc against '_ZTTNSt7__cxx1119basic_istringstreamIcSt11char_traitsIcESaIcEEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(read.o): requires dynamic R_X86_64_PC32 reloc against '_ZTTNSt7__cxx1119basic_istringstreamIcSt11char_traitsIcESaIcEEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(construct.o): requires dynamic R_X86_64_PC32 reloc against '_ZTVSt9basic_iosIcSt11char_traitsIcEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(construct.o): requires dynamic R_X86_64_PC32 reloc against '_ZTTNSt7__cxx1119basic_istringstreamIcSt11char_traitsIcESaIcEEE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(ang.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp8error_noE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(fftcr.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp5mpuu1E' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(fftrc.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp5mpuu1E' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(fft1.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp5mpuu2E' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libarprec.a(fft3.o): requires dynamic R_X86_64_PC32 reloc against '_ZN2mp6mpnrowE' which may overflow at runtime; recompile with -fPIC
collect2: error: ld returned 1 exit status
make[2]: *** [src/shogun/CMakeFiles/shogun.dir/build.make:1475: src/shogun/libshogun.so.18.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:102: src/shogun/CMakeFiles/shogun.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

The `make` step works if ARPREC is uninstalled.

### Architecture and OS
![image](https://user-images.githubusercontent.com/4336207/48887885-f46a3780-ee56-11e8-8ae3-7942988de95b.png)
### Shogun
Git version at:

```bash
 * b5345f577 (HEAD -> develop, origin/develop, origin/HEAD) SG_ADD refactor (#4417)
```

### ARPREC
From the ArchLinux AUR:
![image](https://user-images.githubusercontent.com/4336207/48887967-29768a00-ee57-11e8-9cae-be727b66a665.png)

**Comment by karlnapf** [comment]
Thanks for the report. This should have been fixed a while ago so I dont understand why this appears. We dont have a test build for archlinux though
Do you desperately need ARPREC? That is only used by a few modules within shogun ...

**Comment by HaoZeke** [comment]
I don't actually need ARPREC, so I can make do without it when #4419 works out..


On a side note, would a pull request adding an ArchLinux setup to the CI system be useful?

I can also provide more debugging information if required..

**Comment by karlnapf** [comment]
Thanks for the offer, I think long term maintenance of such a build would be most useful. @vigsterkr I'll leave replying to you. 

As for the example problem. You could try turning them off using `-DBUILD_META_EXAMPLES=Off`, or, if you just want to disable the cpp meta examples `-DTRAVIS_DISABLE_META_CPP=Off`

**Comment by HaoZeke** [comment]
Sure I'd love to maintain one..

Currently I have moved on to Shark but I will switch back if I don't get the eigen errors in my code.

**Comment by ChrisTX** [comment]
The Arch AUR package only builds the static ARPREC library, and that without PIC. As of such, it cannot be linked into `libshogun.so`, as that would require PIC. I've contacted the AUR maintainer with a fixed version of the package, that would enroll ARPREC as a shared library, which is how it should be on Arch.

With the fixed version of the package, I've been able to compile shogun with ARPREC support.
